app/apikey.go

```diff
@@ -167,11 +204,18 @@ func NewAPIKeyCommand(getAPIKeyClientFn GetAPIKeyClientFn) (CommandOut, error) {
 					Action: func(ctx *cli.Context) error {
 						expiry := ctx.Timestamp("expiry")
 						if expiry == nil || expiry.IsZero() {
-							expiryPeriod := ctx.Duration("duration")
-							if expiryPeriod == 0 {
+							expiryPeriod := ctx.String("duration")
+							if expiryPeriod == "" {
+								return fmt.Errorf("no expiry was set")
+							}
+							d, err := parseDuration(expiryPeriod)
+							if err != nil {
+								return fmt.Errorf("failed to parse duration: %v", err)
```

Comment:
Can be good to use `%w` so the error is [wrapped](https://bitfieldconsulting.com/golang/wrapping-errors).